### PR TITLE
Create Helm repos for linkerd2 stable and edge

### PR DESCRIPTION
**Pull request by alpeb**
Fixes linkerd/linkerd2#3365

This adds the Helm repos `https://run.linkerd.io/edge` and
`https://run.linkerd.io/stable`.

So the user would install the stable chart with:
```bash
helm repo add linkerd-stable https://run.linkerd.io/stable
helm install --set blah blah linkerd-stable/linkerd2
```

and the edge chart with:
```bash
helm repo add linkerd-edge https://run.linkerd.io/edge
helm install --set blah blah linkerd-edge/linkerd2
```

Both repos can coexist in the same Helm environment.

I've updated the `install-helm.md` doc accordingly.

The `index.yaml` and `linkerd2-0.1.0.tgz` files were generated with the
`bin/helm -stable|edge` command added in linkerd/linkerd2#3292

### CODE REVIEW

File: linkerd.io/content/dashboard/servicemeshcon-call-for-talks-open-now.md [hunk 1]
```diff
@@ -9,4 +9,4 @@ Using Linkerd? We'd love to help you submit a talk. Get the fame and glory you
 deserve!
 
 [https://events.linuxfoundation.org/events/servicemeshcon-2019/program/call-for-proposals-cfp/](https://events.linuxfoundation.org/events/servicemeshcon-2019/program/call-for-proposals-cfp/
-"https://events.linuxfoundation.org/events/servicemeshcon-2019/program/call-for-proposals-cfp/")
\ No newline at end of file
+"https://events.linuxfoundation.org/events/servicemeshcon-2019/program/call-for-proposals-cfp/")
```
Comment (cpretzer) [inline linkerd.io/content/dashboard/servicemeshcon-call-for-talks-open-now.md]: These CFPs are closed, so do we need to add this?

**Comment by ihcsim** [comment]
Combining what I gathered from https://github.com/BuoyantIO/buoyant-internal/pull/1023, https://github.com/linkerd/linkerd2/pull/3292 and this PR, my understanding is that we are publishing the chart `.tgz` file for every edge and stable release, by pushing them to this repo. And in order to do this, we need to change the chart version for every release, in order to generate a different `.tgz` chart file, right?

TL;DR:

1. Can we instead only publish one chart for the edge, and not bother with changing the chart version?
1. For stable releases, we can change the chart release to match the Linkerd release version.

Some details on chart publication and user's install/upgrade workflow:

The reason behind the proposal to publish only one chart for the edge is because I don't want try to figure out which edge release does chart `0.0.1` correspond to. It will also remove the need to update the chart version on every edge release.

E.g., during edge release, we modified the `appVersion` to reflect the edge version, but keep `version` as a constant:
```
✗ helm package --app-version edge-19.9.2 --version 0.0.1+edge --destination charts/ charts/linkerd2
✗ helm repo index --url http://127.0.0.1:7000/charts/edge charts/edge

✗ helm package --app-version edge-19.9.3 --version 0.0.1+edge --destination charts/ charts/linkerd2
✗ helm repo index --url http://127.0.0.1:7000/charts/edge charts/edge
```

I imagine this is what the user's workflow will look like:
```
# using my local helm repo
✗ helm repo add linkerd-edge http://127.0.0.1:7000/charts/edge                                                
"linkerd-edge" has been added to your repositories

✗ helm repo list
NAME            URL
stable          https://kubernetes-charts.storage.googleapis.com
local           http://127.0.0.1:8879/charts
istio.io        https://storage.googleapis.com/istio-prerelease/daily-build/master-latest-daily/charts
linkerd-edge    http://127.0.0.1:7000/charts/edge

✗ helm repo update   
...
Update Complete.

# user discovered edge-19.9.1
✗ helm search linkerd2                                                                                       
NAME                    CHART VERSION   APP VERSION     DESCRIPTION
linkerd-edge/linkerd2   0.0.1+edge      edge-19.9.1     Linkerd gives you observability, reliability, and securit...

# install edge-19.9.1
✗ helm install # etc...                   

✗ helm list                                                                                                  
NAME            REVISION        UPDATED                         STATUS          CHART                   APP VERSION     NAMESPACE
linkerd2        1               Sun Sep  8 20:05:00 2019        DEPLOYED        linkerd2-0.0.1+edge     edge-19.9.1     default

# update repo to get edge-19.9.2
✗ helm repo update   
...
Update Complete.

# user discovered edge-19.9.2
✗ helm search linkerd2                                                                                       
NAME                    CHART VERSION   APP VERSION     DESCRIPTION
linkerd-edge/linkerd2   0.0.1+edge      edge-19.9.2     Linkerd gives you observability, reliability, and securit...

# upgrade to edge-19.9.2
✗ helm upgrade linkerd2 linkerd-edge/linkerd2
# etc...

✗ helm list                                                      
NAME            REVISION        UPDATED                         STATUS          CHART                   APP VERSION     NAMESPACE
linkerd2        2               Sun Sep  8 20:20:11 2019        DEPLOYED        linkerd2-0.0.1+edge     edge-19.9.2     default
```

As for stable releases, how about we make the chart version the same as the release itself? I find the default chart versioning of 0.0.1, 0.0.2 etc. meaningless.
```
✗ helm package --app-version=stable-2.5.0 --version=2.5.0 --destination charts/stable charts/linkerd2
Successfully packaged chart and saved it to: charts/stable/linkerd2-2.5.0.tgz
✗  helm repo index --url http://127.0.0.1:7000/charts/stable charts/stable

✗ helm package --app-version=stable-2.6.0 --version=2.6.0 --destination charts/stable charts/linkerd2     
Successfully packaged chart and saved it to: charts/stable/linkerd2-2.6.0.tgz
✗  helm repo index --url http://127.0.0.1:7000/charts/stable charts/stable
```

On the user's side:
```
✗ helm repo add linkerd http://127.0.0.1:7000/charts/stable
"linkerd" has been added to your repositories

✗ helm repo list                                                                                             
NAME            URL
stable          https://kubernetes-charts.storage.googleapis.com
local           http://127.0.0.1:8879/charts
istio.io        https://storage.googleapis.com/istio-prerelease/daily-build/master-latest-daily/charts
linkerd-edge    http://127.0.0.1:7000/charts/edge
linkerd         http://127.0.0.1:7000/charts/stable

# user discovered stable-2.5
✗ helm search linkerd2                                                                                       
NAME                    CHART VERSION   APP VERSION     DESCRIPTION
linkerd-edge/linkerd2   0.0.1+edge      edge-19.9.2     Linkerd gives you observability, reliability, and securit...
linkerd/linkerd2        2.5.0           stable-2.5.0    Linkerd gives you observability, reliability, and securit...

# install stable-2.5
✗ helm install # etc..

✗ helm list                                                                                               
NAME            REVISION        UPDATED                         STATUS          CHART           APP VERSION     NAMESPACE
linkerd2        1               Sun Sep  8 20:39:01 2019        DEPLOYED        linkerd2-2.5.0  stable-2.5.0    default

✗ helm repo update 

# user discovered stable-2.6.0
✗ helm search linkerd2           
NAME                    CHART VERSION   APP VERSION     DESCRIPTION
linkerd-edge/linkerd2   0.0.1+edge      edge-19.9.2     Linkerd gives you observability, reliability, and securit...
linkerd/linkerd2        2.6.0           stable-2.6.0    Linkerd gives you observability, reliability, and securit...

# upgrade to stable-2.6.0
✗ helm upgrade linkerd2 linkerd/linkerd2 

✗ helm list                                                                                                 
NAME            REVISION        UPDATED                         STATUS          CHART           APP VERSION     NAMESPACE
linkerd2        2               Sun Sep  8 20:51:39 2019        DEPLOYED        linkerd2-2.6.0  stable-2.6.0    default
```

File: linkerd.io/content/dashboard/servicemeshcon-call-for-talks-open-now.md (hunk 1 above)
Comment (alpeb) [reply linkerd.io/content/dashboard/servicemeshcon-call-for-talks-open-now.md]: The change consisted just on adding a newline at the end of the file to make the linter pass. As for leaving that file, I guess we'd like to leave everything there for historical purposes, but I could be wrong.

**Comment by alpeb** [comment]
@ihcsim Regarding edge charts, I'm a little uneasy about having the same chart version not being deterministic about what linkerd version it corresponds to, plus we wouldn't have an archive of past charts. If a user reports a problem with a chart then we'd have to re-generate it. Also it seems that wouldn't play well with Helm hub where each chart version gets its own page, for example:
https://hub.helm.sh/charts/flagger/flagger/0.18.1
I believe the upgrade flow would be the same as what you describe (in my testing though I didn't need to call `helm repo update` to grab a new chart, but it is necessary to have `helm search` perform its thing using the latest chart metadata)

That being said, we could indeed have the chart version match the app version, with some minor changes to make it semver.

**Comment by ihcsim** [comment]
Per post stand-up convo, we will publish the chart for every edge and stable release, to a new GCS bucket. We will automate this by adding the `bin/helm-build` script to GitHub Actions.

File: linkerd.io/content/2/tasks/install-helm.md [hunk 2]
```diff
@@ -43,25 +40,29 @@ helm install \
...
 ## Setting High-Availability
 
-Also under `charts/linkerd2` there's a file `values-ha.yaml` that overrides some
+The chart contains a file `values-ha.yaml` that overrides some
 default values as to set things up under a high-availability scenario, analogous
 to the `--ha` option in `linkerd install`. Values such as higher number of
 replicas, higher memory/cpu limits and affinities are specified in that file.
 
-As usual the `-f` flag to provide the override file, for example:
+You can get ahold of `values-ha.yaml` by fetching the chart files:
```
Comment (ihcsim) [inline linkerd.io/content/2/tasks/install-helm.md]: Or just link to the source on GH.

Comment (alpeb) [reply linkerd.io/content/2/tasks/install-helm.md]: The source on GH (master) might not be in sync with the the Helm release in `https://helm.linkerd.io/(stable|edge)`, so I'd prefer to have the user extract it from the release itself.
